src/panels/CreateClusterPanel.ts

```diff
@@ -444,5 +444,13 @@ function getServicePrincipalId(result: AzureAuthenticationSession): string {
     if (!result || !result.account || !result.account.id) {
         return "";
     }
-    return result.account.id.split("/")[1];
+    // account id is in the format of tenantID/servicePrincipalId
+    // sometimes in case of non aad auth it is in the format of servicePrincipalId.tenantID
+    if (result.account.id.includes("/")) {
```

Comment:
💡 small improvement, sorry for delay, (due to on-call stuff) <3 :

```*.ts
    const accountId = result.account.id;

    // account id is in the format of tenantID/servicePrincipalId
    // sometimes in case of non-AAD auth, it is in the format of servicePrincipalId.tenantID
    if (accountId.includes("/")) {
        return accountId.split("/")[1];
    } else if (accountId.includes(".")) {
        return accountId.split(".")[0];
    } else {
        return "";
    }
```

OR if you feel brave: and want to improve / test here is second option :) 

```.ts
function getServicePrincipalId(result: AzureAuthenticationSession | null | undefined): string {
    // Ensure result and necessary fields are valid before processing
    if (!result?.account?.id) {
        return "";
    }

    const accountId = result.account.id;

    // account id is in the format of tenantID/servicePrincipalId
    // sometimes in case of non-AAD auth, it is in the format of servicePrincipalId.tenantID
    if (accountId.includes("/")) {
        return accountId.split("/")[1];
    } else if (accountId.includes(".")) {
        return accountId.split(".")[0];
    } else {
        return "";
    }
}
```

Reply:
updated @Tatsinnit , approvals please.